lib/substitutions.js

```diff
@@ -20,6 +22,22 @@ add("&lt;", "<");
 add("&gt;", ">");
 add("&amp;", "&"); // &amp; must be after all replacements involving &s.
 
+var emojiIndex = (function indexEmoji() {
+    var emojiJson = fs.readFileSync("./external/emoji-data/emoji.json");
+    var emoji = JSON.parse(emojiJson);
+    var index = {};
+    for (var i = 0; i < emoji.length; ++i) {
+        var e = emoji[i];
+        for (var j = 0; j < e.short_names.length; ++j) {
+            if (e["unified"].indexOf("-") < 0) {
+                var char = String.fromCodePoint(parseInt(e["unified"], 16));
+                index[":" + e.short_names[j] + ":"] = char;
+            }
+        }
+    }
+    return index;
+})();
```

Comment:
Avoiding polluting the file namespace with random stuff.

In particular, I got complaints from the linter that "i" was already defined in file scope when I didn't have this hidden away in a function.

I agree that the function doesn't need a name, though... :)